### DISCUSSION TITLE
Download only one preview image

### DISCUSSION THREAD

**Discussion by Zaazu** [BlafKing/sd-civitai-browser-plus#64] [Ideas]
Hey, 

When downloading a model, it doesn't automatically download the preview image, it would be nice if it just downloaded that one image, rather than all of the images via clicking on the download images button. Right now I'm clicking on the download all button, then deleting the excess.

Cheers :)

**Comment by BlafKing** [comment]
That's odd, it should already automatically download the preview image when you're downloading a model, I will check in a bit when I'm back home to see if any updates broke that function :')
Could you send the model which didn't automatically download a preview image? Just in case it was only not working for that model
No need if it doesn't download the preview for every model :)

**Comment by Zaazu** [comment]
It hasn't been doing it for any model or lora, I had to click the download images button

**Comment by BlafKing** [comment]
I just checked an it's working fine for me, automatically downloads a preview image when downloading a model.

https://github.com/BlafKing/sd-civitai-browser-plus/assets/9644716/91586a02-38d5-4ff8-942a-c79c337a68f5

Could you provide an example model which didn't download any preview images for you so that I can double check on that model?
Also please make sure you're using the latest version :)

**Comment by Zaazu** [comment]
I'll check and get back to you tonight, cheers

**Comment by Zaazu** [comment]
I downloaded the same lora you did in the video and it doesn't download the preview.
However, checking the console there's an error. I set the lora path via a command line argument, so maybe that's the issue. The model downloads fine but not the preview, so maybe there's a bug in the code.

![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/22891753/861cf6c2-84ac-44c6-a520-f887934fe539)
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/22891753/0b31de1d-f077-4db7-82ba-e258e6fe9d29)

**Comment by BlafKing** [comment]
I see now that I've been passing the wrong argument to the `save_preview` function which is likely causing the issue, I've published a new update for it.
Please let me know if this has fixed your issue! :)

**Comment by Zaazu** [comment]
It's fixed the issue, thanks :D
